### PR TITLE
Theming - Postcss plugin to process URLs

### PR DESCRIPTION
**Pull request by MidhunSureshR**
This plugin replaces each url in the css with the result of running a callback. 
We'll use the callback to colorize svgs

### CODE REVIEW

File: scripts/postcss/svg-colorizer.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Builds a new svg with the colors replaced and returns its location.
+ * @param {string} svgLocation The location of the input svg file
+ * @param {string} primaryColor Primary color for the new svg
+ * @param {string} secondaryColor Secondary color for the new svg
+ */
+module.exports.buildColorizedSVG = function (svgLocation, primaryColor, secondaryColor) {
+    const svgCode = fs.readFileSync(svgLocation, { encoding: "utf8"});
+    let coloredSVGCode = svgCode.replaceAll("#ff00ff", primaryColor);
+    coloredSVGCode = coloredSVGCode.replaceAll("#00ffff", secondaryColor);
```
Comment (bwindels) [inline scripts/postcss/svg-colorizer.js:28]: can we throw an error here if nothing was replaced for both the primary and secondary color? It probably means you got the hardcoded colors wrong (e.g. uppercased) in your SVG.

File: scripts/postcss/css-url-processor.js [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        throw new Error(`Variable ${secondary} not found in resolved color variables!`);
+    }
+    return [primaryColor, secondaryColor];
+}
+
+function processURL(decl, replacer, colorMap) {
+    const value = decl.value;
+    const parsed = valueParser(value);
+    parsed.walk(async node => {
+        if (node.type !== "function" || node.value !== "url") {
```
Comment (bwindels) [inline scripts/postcss/css-url-processor.js:42]: I thought the urls would already be replaced with a variable at this point by [the other plugin](https://github.com/vector-im/hydrogen-web/pull/703/files#diff-edd3a0e0c14c7f3a611b4b972dcc1d9e86be041945df96c9fab21ab87e264a31R40)?

Comment (MidhunSureshR) [reply scripts/postcss/css-url-processor.js:42]: Right, the URLs would be in the :root section like:
```css
:root {
    --icon-url-1: url('./foo/bar/../test.svg?primary=accent-color);
}
```
The code is trying to find these URL values.

Comment (bwindels) [reply scripts/postcss/css-url-processor.js:42]: Oh, right, that makes sense!

**Review by bwindels — APPROVED** [review]

**Pull request merged**